### PR TITLE
TEL-5733

### PR DESCRIPTION
**Pull request by damirn**
Please read my [analysis](https://telnyx.atlassian.net/browse/TEL-5733) too.

### CODE REVIEW

File: mod_janus.c [hunk 1]
```diff
@@ -427,9 +427,9 @@ static void *SWITCH_THREAD_FUNC server_thread_run(switch_thread_t *pThread, void
 						switch_assert(channel);
 						//NB. the server is likely to have been removed by the time the hangup has completed
 						switch_channel_hangup(channel, SWITCH_CAUSE_DESTINATION_OUT_OF_ORDER);
+						(void) hashDelete(&pServer->senderIdLookup, tech_pvt->senderId);
 						switch_core_session_rwunlock(session);
 					}
-					(void) hashDelete(&pServer->senderIdLookup, tech_pvt->senderId);
```
Comment (dev-ryanc) [inline mod_janus.c:432]: Another potential crash is due to session is no longer exist when locating the uuid hence the crash is due to null pointer tech_pvt accessing senderId

File: mod_janus.c [hunk 2]
```diff
@@ -427,9 +427,9 @@ static void *SWITCH_THREAD_FUNC server_thread_run(switch_thread_t *pThread, void
 						switch_assert(channel);
 						//NB. the server is likely to have been removed by the time the hangup has completed
 						switch_channel_hangup(channel, SWITCH_CAUSE_DESTINATION_OUT_OF_ORDER);
+						(void) hashDelete(&pServer->senderIdLookup, tech_pvt->senderId);
```
Comment (dev-ryanc) [inline mod_janus.c:430]: If in case the session didn't exist then it is also possible that we no longer delete the hash recod in `senderIdLookup`

**Review by dev-ryanc — CHANGES_REQUESTED** [review]

**Comment by damirn** [comment]
Please re-review; @dev-ryanc'c comments were addressed

File: hash.c [hunk 3]
```diff
@@ -92,23 +92,27 @@ void *hashFind(const hash_t *pHash, const janus_id_t id) {
   return pResult;
 }
 
-switch_status_t hashDelete(const hash_t *pHash, const janus_id_t id) {
+switch_status_t hashDeleteStr(const hash_t *pHash, char *pIdStr) {
```
Comment (dev-ryanc) [inline hash.c:95]: In hash header, it supposed to be have this two function
- hashDelete
- hashDeleteStr

but this change is hashDeleteStr overwrites the hashDelete.

**Review by dev-ryanc — CHANGES_REQUESTED** [review]

File: hash.c (hunk 3 above)
Comment (damirn) [reply hash.c:95]: i think it is the artifact of the unified diff, both functions are there

**Review by minhtuan1407-telnyx — APPROVED** [review]

**Pull request merged**